src/main/java/com/shopping_app/shoppingApp/service/OrderService.java

```diff
@@ -0,0 +1,145 @@
+    private final AddressService addressService;
+    private final CartItemRepository cartItemRepository;
+    private final ProductRepository productRepository;
+
+    public List<OrderResponse> getUserOrders(Long userId) {
+        Optional<List<Order>> orders = orderRepository.findAllByUserId(userId);
+        return orders.get().stream().map(OrderResponse::from).collect(Collectors.toList());
+    }
+
+    public OrderResponse getUserOrderById(Long userId, Long orderId) {
+        return OrderResponse.from(getOrderById(userId, orderId));
+    }
+
+    public OrderResponse addOrders(Long userId, OrderAddRequest orderAddRequest) {
+
+        Cart cart = cartService.getCartByUserId(userId);
+        cartService.validateCart(userId, orderAddRequest.getCartId());
+
+        Address address = addressService.getAddressById(userId, orderAddRequest.getShippingAddressId());
+        User user = userService.getUserById(userId);
+
+        Order order = new Order();
+        Set<OrderItems> orderItems = addOrderItems(cart.getCartItems(), order);
+
+        order.setOrderItems(orderItems);
+        order.setStatus(OrderStatus.BOOKED); // Default
+        order.setShippingAddress(address);
+        order.setDeliveryDate(LocalDate.now().plusDays(3)); // adding three days by default
+        order.setUser(user);
+        order.setTrackingNumber(UUID.randomUUID().toString());
+
+        Order savedOrder = orderRepository.save(order);
+
+        // once order placed , do-- cart empty and update inventory count of product
+        updateProductAvailability(savedOrder);
+        removeItemsFromCart(cart.getCartItems());
+
+        return OrderResponse.from(savedOrder);
+    }
+
+    public void updateProductAvailability(Order order) {
+        for (OrderItems orderItems : order.getOrderItems()) {
+            Product product = orderItems.getProduct();
+            product.setAvailableQuantity(product.getAvailableQuantity() - orderItems.getQuantity());
+            productRepository.save(product);
+        }
+    }
+
+    public Set<OrderItems> addOrderItems(Set<CartItem> cartItems, Order order) {
+        Set<OrderItems> orderItems = new HashSet<>();
+
+        for (CartItem e : cartItems) {
+            // Check product availability
+            cartService.checkProductAvailability(e.getProduct(), e.getQuantity());
+
+            OrderItems orderItem = new OrderItems();
+            orderItem.setQuantity(e.getQuantity());
+            orderItem.setProduct(e.getProduct());
+            orderItem.setTotalPrice((e.getProduct().getPrice() * e.getQuantity()));
+            orderItem.setOrder(order);
+            orderItems.add(orderItem);
+        }
+        return orderItems;
+    }
+
+    public OrderResponse updateUserOrderStatus(Long userId, Long orderId, OrderUpdateRequest orderUpdateRequest) {
+        Order order = getOrderById(userId, orderId);
+        checkOrderValidity(order);
+
+        order.setStatus(orderUpdateRequest.getOrderStatus());
```

Comment:
I can send any status from the api and it will be updated for the order?

Reply:
Yes it will be , will add check here  , missed out this check

Reply:
added this new logic check